### PR TITLE
Not Null and Unique Constraints

### PR DESCRIPTION
**Pull request by signedav**
I thought it's a bug, but apparently it has not been implemented before.

Now Not Null and Unique Constraints are supported.
WhatsApp Image 2020-03-25 at 10 50 07

The descriptionmessage is the list of all the messages including the ones generated for not null and unique. They are not separated by it's strength. This could be improved but is actually how QGIS behaves as well.

### CODE REVIEW

**Comment by signedav** [comment]
Even if it's no bug, I suggest to put it into the 1.5 release since we advertise there the cool soft constraints.

**Comment by qfield-fairy** [comment]
Uploaded test apks for
 - [**armv7**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-906-31a0c6ed04c676b35d5155aa6f74fbebe1ee2514-armv7.apk)
 - [**x86**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-906-31a0c6ed04c676b35d5155aa6f74fbebe1ee2514-x86.apk)
 - [**arm64_v8a**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-906-31a0c6ed04c676b35d5155aa6f74fbebe1ee2514-arm64_v8a.apk)
 - [**x86_64**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-906-31a0c6ed04c676b35d5155aa6f74fbebe1ee2514-x86_64.apk)

File: src/qml/FeatureForm.qml [hunk 1]
```diff
@@ -242,7 +242,8 @@ Page {
           top: fieldLabel.bottom
         }
 
-        text: !ConstraintHardValid ? qsTr( "Error: " ) + ConstraintDescription : !ConstraintSoftValid ? qsTr( "Warning: " ) + ConstraintDescription : ''
+        //text: !ConstraintHardValid ? qsTr( "Error: " ) + ConstraintDescription : !ConstraintSoftValid ? qsTr( "Warning: " ) + ConstraintDescription : ''
```
Comment (m-kuhn) [inline src/qml/FeatureForm.qml]: Can you kill this line? It's NvPractice to comment, OgPractice is to trust the git log ;)

File: src/core/attributeformmodelbase.cpp [hunk 2]
```diff
@@ -289,15 +290,28 @@ void AttributeFormModelBase::flatten( QgsAttributeEditorContainer *container, QS
...
-        if ( !field.constraints().constraintExpression().isEmpty() )
+        // create constraint description
+        QStringList descriptions;
+        if( !field.constraints().constraintDescription().isEmpty() )
+        {
+          descriptions << field.constraints().constraintDescription();
+        }
+        if ( field.constraints().constraints() & QgsFieldConstraints::ConstraintNotNull )
+        {
+          descriptions << tr("Not NULL");
```
Comment (m-kuhn) [inline src/core/attributeformmodelbase.cpp]: Can you enable your pre-commit hook and run this file through astyle?

File: src/qml/FeatureForm.qml (hunk 1 above)
Comment (signedav) [reply src/qml/FeatureForm.qml]: Oh, I just forgot.

File: src/core/androidplatformutilities.cpp [hunk 3]
```diff
@@ -260,6 +260,7 @@ bool AndroidPlatformUtilities::checkAndAcquirePermissions( const QString &permis
 
 void AndroidPlatformUtilities::setScreenLockPermission( const bool allowLock )
 {
+  /*
```
Comment (m-kuhn) [inline src/core/androidplatformutilities.cpp]: That, my friend, was not meant for this pull request!

**Comment by qfield-fairy** [comment]
Uploaded test apks for
 - [**armv7**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-906-172171bdcd8191165f86110a053770386b5c142a-armv7.apk)
 - [**x86**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-906-172171bdcd8191165f86110a053770386b5c142a-x86.apk)
 - [**arm64_v8a**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-906-172171bdcd8191165f86110a053770386b5c142a-arm64_v8a.apk)
 - [**x86_64**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-906-172171bdcd8191165f86110a053770386b5c142a-x86_64.apk)

**Comment by qfield-fairy** [comment]
Uploaded test apks for
 - [**armv7**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-906-066cb4777821bb55f82daad9ff24ce7b34633476-armv7.apk)
 - [**x86**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-906-066cb4777821bb55f82daad9ff24ce7b34633476-x86.apk)
 - [**arm64_v8a**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-906-066cb4777821bb55f82daad9ff24ce7b34633476-arm64_v8a.apk)
 - [**x86_64**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-906-066cb4777821bb55f82daad9ff24ce7b34633476-x86_64.apk)

File: src/core/androidplatformutilities.cpp (hunk 3 above)
Comment (signedav) [reply src/core/androidplatformutilities.cpp]: Argh... not the first time happening.

**Comment by qfield-fairy** [comment]
Uploaded test apks for
 - [**armv7**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-906-c119f0ad07d30816739e7ce0b8ed43e33c606598-armv7.apk)
 - [**x86**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-906-c119f0ad07d30816739e7ce0b8ed43e33c606598-x86.apk)
 - [**arm64_v8a**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-906-c119f0ad07d30816739e7ce0b8ed43e33c606598-arm64_v8a.apk)
 - [**x86_64**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-906-c119f0ad07d30816739e7ce0b8ed43e33c606598-x86_64.apk)

**Pull request merged**